From 9219fcc0d2265d84971da4bdc702367ac4708863 Mon Sep 17 00:00:00 2001
From: Gao Pan <b54642@freescale.com>
Date: Tue, 1 Sep 2015 13:14:08 +0800
Subject: [PATCH 0961/5242] MLK-11472 sim: imx: set sim1 IOMUX to default
 state

commit  f82631bda91c8e0b6ff55aebaf0f677802f9b9c0 from
https://source.codeaurora.org/external/imx/linux-imx.git

SIM1 IOMUX is changed into reset state in LPSR mode.As a result,
sim can't work again.

This patch sets sim1 IOMUX to default state after existing from LPSR mode.

(cherry-picked from commit 8e237775cd413645bfd806e4c648954e1a773a2d)

Signed-off-by: Gao Pan <b54642@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/sim/imx_sim.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mxc/sim/imx_sim.c b/drivers/mxc/sim/imx_sim.c
index 7b00e93..ba13b97 100755
--- a/drivers/mxc/sim/imx_sim.c
+++ b/drivers/mxc/sim/imx_sim.c
@@ -1809,6 +1809,8 @@ static int sim_suspend(struct platform_device *pdev, pm_message_t state)
 	if (sim->open_cnt)
 		clk_disable_unprepare(sim->clk);
 
+	pinctrl_pm_select_sleep_state(&pdev->dev);
+
 	return 0;
 }
 
@@ -1819,6 +1821,8 @@ static int sim_resume(struct platform_device *pdev)
 	if (sim->open_cnt)
 		clk_prepare_enable(sim->clk);
 
+	pinctrl_pm_select_default_state(&pdev->dev);
+
 	return 0;
 }
 #else
-- 
1.7.9.5

